src/test/java/hudson/tasks/junit/pipeline/JUnitResultsStepTest.java

```diff
@@ -385,14 +368,57 @@ public void skipBuildUnstable() throws Exception {
                 "    assert currentBuild.result == null\n" +
                 "  }\n" +
                 "}\n", true));
-        FilePath ws = rule.jenkins.getWorkspaceFor(j);
-        FilePath testFile = ws.child("test-result.xml");
-        testFile.copyFrom(JUnitResultsStepTest.class.getResource("junit-report-testTrends-first-2.xml"));
+        copyToWorkspace(j, JUnitResultsStepTest.class.getResource("junit-report-testTrends-first-2.xml"), "test-result.xml");
         WorkflowRun r = rule.waitForCompletion(j.scheduleBuild2(0).waitForStart());
         rule.assertBuildStatus(Result.SUCCESS, r);
         assertStageResults(r, 1, 8, 3, "first");
     }
 
+    @Test
+    public void ageResetSameTestSuiteName() throws Exception {
+        WorkflowJob j = rule.jenkins.createProject(WorkflowJob.class, "currentBuildResultUnstable");
+        j.setDefinition(new CpsFlowDefinition("stage('stage 1') {\n" +
+                "  node {\n" +
+                "    junit(testResults: '*-1.xml')\n" +
+                "  }\n" +
+                "}\n" +
+                "stage('stage 2') {\n" +
+                "  node {\n" +
+                "    junit(testResults: '*-2.xml')\n" +
+                "  }\n" +
+                "}\n", true));
+        copyToWorkspace(j, JUnitResultsStepTest.class.getResource("ageReset-1.xml"), "ageReset-1.xml");
+        copyToWorkspace(j, JUnitResultsStepTest.class.getResource("ageReset-2.xml"), "ageReset-2.xml");
+        WorkflowRun r = rule.waitForCompletion(j.scheduleBuild2(0).waitForStart());
+        rule.assertBuildStatus(Result.UNSTABLE, r);
+        CaseResult caseResult = findCaseResult(r, "aClass.methodName");
+        assertNotNull(caseResult);
+        assertEquals(1, caseResult.getAge());
+
+        // Run a second build, age should increase
+        r = rule.waitForCompletion(j.scheduleBuild2(0).waitForStart());
+        rule.assertBuildStatus(Result.UNSTABLE, r);
+        caseResult = findCaseResult(r, "aClass.methodName");
+        assertNotNull(caseResult);
+        assertEquals(2, caseResult.getAge());
```

Comment:
Is this what fails with `src/main/` reverted? Not obvious to me how the test case is verifying that there are two suites.